src/common/cockpitpipe.c

```diff
@@ -1505,6 +1529,47 @@ cockpit_pipe_get_stderr (CockpitPipe *self)
   return priv->err_buffer;
 }
 
+gchar *
+cockpit_pipe_take_stderr_as_utf8 (CockpitPipe *self)
+{
+  GByteArray *buffer;
+  GBytes *bytes;
+  GBytes *clean;
+  gchar *data;
+  gsize length;
+
+  buffer = cockpit_pipe_get_stderr (self);
+  if (!buffer)
+    return NULL;
+
+  /* A little more complicated to avoid big copies */
+  g_byte_array_ref (buffer);
+  g_byte_array_append (buffer, (guint8 *)"x", 1); /* place holder for null terminate */
+  bytes = g_byte_array_free_to_bytes (buffer);
```

Comment:
...and because of the extra ref, this will indeed take a copy.

Reply:
I don't think so... this ultimately calls `g_array_free (buffer, FALSE)`, which will always take the actual memory away from `buffer` but might preserve the `buffer` wrapper itself.  The `g_byte_array_ref` above makes sure that the wrapper stays intact.

Reply:
Ah.  You're right.  This also invalidates my above comments about how this isn't a `take` anymore.  It clearly is.

I read the docs incorrectly.